docs/features/source-generators.md

```diff
@@ -0,0 +1,291 @@
+generation, and can be completely disabled, if required. The IDE will work on in-memory 
+copies of the generated source texts (for 'Find all references', breakpoints etc.) and 
+periodically flush any changes to disk. 
+
+To support the use case where a user wishes to generate the source text, then commit 
+the generated files to source control, we will allow changing the location of the 
+generated files via an appropriate command line switch, and matching MSBuild property 
+(naming still to be determined).
+
+In these cases it will be up to the user if they wish to generate over the files again
+in the future (in which case they would still be generated, but output to a 
+source controlled location), or remove the generators and perform the action as a one 
+time step. 
+
+It is currently an open question how for example, the action of setting a breakpoint in
+a disk-based generated file will function. 
+
+TK: how do we save PDBs/Source link etc?
+
+### Editing experiences for third party languages
+
+One of the interesting scenarios that source generators will enable is essentially 
+the 'embedding' of C# within other languages (and vice versa). This is how Razor 
+works today, and the Razor team maintains a significant language service investment 
+in Visual Studio to enable it. 
+
+A possible goal of this project would be to find a generic way to represent this:
+that would allow the Razor team to reduce their tooling investment, while allowing
+third parties the opportunity to enable the same sort of experiences 
+(including 'Go to definition', 'Find all references' etc.) relatively cheaply.
+
+The current thinking is to have some form of 'side-channel' available to
+the generator. As the generator emits source text, it would indicate where 
+in the original document this was generated from. This would allow the 
+compiler API to track e.g. a generated `Symbol` as having an `OriginalDefinition` 
+that represents a span of third party source text (such as a Razor tag in a
+`.cshtml` file). 
+
+We discussed embedding this directly in the source text via `#pragma` but 
+this would require language changes and limit the feature to a specific version
+of C#. Other considerations could be specially formed comments or `#if FALSE --`
+blocks. In general a 'side-channel' approach seems preferable to specially crafted
+grammar in the generated text. 
+
+This is not necessarily a goal required for the success of Source Generators;
+Razor’s language service can be updated to work with source generators 
+if it proves to be infeasible, but it certainly something we want to consider 
+as part of the work.
+
+### MSBuild Integration
+
+It is expected that generators will need some form of configuration system, and we intend to allow 
+certain properties to flow through from MSBuild to facilitate this.
+
+> **Note**: This is still under design and open to change.
+
+
+### Performance targets
```

Comment:
Should we consider out of process from the start given the nature of generating very very large sources, even if the generator was written well?

Reply:
For IDE we're looking at a minimal bring up scenario to allow the compiler work to proceed. It's not intended to be the final shipping solution. This reduced their cost significantly and unblocked our ability to move the compiler forward. Expectation is that for shipping we'll move it out of proc just as we do for analyzers.

Reply:
Much yeet.